### DISCUSSION TITLE
VITS fine tuning question

### DISCUSSION THREAD

**Discussion by weidezhang** [coqui-ai/TTS#2457] [General Q&A]
Hi 

I have a specific question regarding the VITS model fine tuning. I used about 3k samples (about 1 hour recording) extracted from some story telling audio from the same speaker. After around 1k epoches, it seems the word from the testing senstence is clear but still sound  a little weird.  Since I was doing fine tuning, I adjusted my initial learning rate a little lower to 0.0005 for lr_gen, lr_disc to 0.0005. 

And also it seems after 1k epoches, the avg_loss_1 are not going down as well.  I wonder what will be the next steps to improve the naturalness of the speaker ? 

1. shall i use speaker embedding and add speaker embedding loss ? will that improve the naturalness ?  
2. shall i adjust the loss weight and freeze some layer such as text encoder layer? will that help ? 
3. adding more data, i guess for sure, but so far it is not cheap to get more data.

**Comment by p0p4k** [comment]
Hi, how many steps is in your one epoch? Try freezing few layers and playing around with the training. What is your value of loss_1 and mel_loss at the end of 1k epochs?

**Reply by p0p4k** [reply]
I think the model needs more training though, if you are changing any characters or punctuation while restoring/continuing from this model_ckpt, remember to delete the phoneme_cache folder first.

**Reply by weidezhang** [reply]
"> Hi, how many steps is in your one epoch? Try freezing few layers and playing around with the training. What is your value of loss_1 and mel_loss at the end of 1k epochs?"


my batch size is 32 and group size is 5. total steps for each epoch is 90. Attached is the screenshot for my loss. As you can see the loss_1 and mel_loss seems going backup after epoch 222. 20k / 90 from the diagram. does it looks like an overfitting ? 

 
![Screenshot from 2023-03-26 08-58-23](https://user-images.githubusercontent.com/455617/227780916-755e3115-92ea-4fce-b547-3aeb93d03c95.png)

**Comment by p0p4k** [comment]
I will not say it is overfitting exactly. Cause if we notice the values (example, loss_1), the value are kind of stagnant, (24 to 26 is not a very big change). I am training a vits model right now with multi speaker and it has similar values like yours. I might increase the size of some model's layers/blocks and try again.

**Reply by p0p4k** [reply]
Batch size of 32 is ok. 
- https://twitter.com/ylecun/status/989610208497360896?lang=en 
- https://arxiv.org/abs/1804.07612
